packages/odyssey-react-mui/src/FieldComponentProps.ts

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+export type FieldComponentProps = {
+  /**
+   * If `error` is not undefined, the `input` will indicate an error.
+   */
+  errorMessage?: string;
+  /**
+   * The helper text content.
+   */
+  hint?: string;
+  /**
+   * The id of the `input` element.
+   */
+  id?: string;
+  /**
+   * If `true`, the component is disabled.
+   */
+  isDisabled?: boolean;
+  /**
+   * If `true`, the `input` element is not required.
+   */
+  isOptional?: boolean;
+  /**
+   * It prevents the user from changing the value of the field
+   */
+  isReadOnly?: boolean;
+  /**
+   * The name of the `input` element. Defaults to the `id` if not set.
+   */
+  name?: string;
+};
```

Comment:
Can we include the Storybook metadata in this file as well? That way, we don't have to duplicate that in the Storybook part as well.

Reply:
Not sure what you mean exactly here. Just create a new object with the data like 
```js
export const fieldComponentPropsMetaData {
  errorMessage: {
      control: "text",
      description:
        "If `error` is not undefined, the `input` will indicate an error",
      table: {
        type: {
          summary: "string",
        },
      },
    },
    ...
```
And import that into each story?